homeassistant/components/sensor/logi_circle.py

```diff
@@ -0,0 +1,154 @@
+        self._camera = camera
+        self._icon = 'mdi:{}'.format(SENSOR_TYPES.get(self._sensor_type)[2])
+        self._name = "{0} {1}".format(
+            self._camera.name, SENSOR_TYPES.get(self._sensor_type)[0])
+        self._state = STATE_UNKNOWN
+        self._tz = str(hass.config.time_zone)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            'battery_saving_mode': ('on' if self._camera.battery_saving
+                                    else 'off'),
+            'firmware': self._camera.firmware,
+            'generation': self._camera.model_generation,
+            'ip_address': self._camera.ip_address,
+            'mac_address': self._camera.mac_address,
+            'microphone_gain': self._camera.microphone_gain,
+            'model': self._camera.model,
+            'mount': self._camera.mount,
+            'plan': self._camera.plan_name,
+            'speaker_volume': self._camera.speaker_volume,
+            'timezone': self._camera.timezone
+        }
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        if (self._sensor_type == 'battery_level' and
+                self._state != STATE_UNKNOWN):
+            return icon_for_battery_level(battery_level=int(self._state),
+                                          charging=False)
+        if (self._sensor_type == 'privacy_mode' and
+                self._state != STATE_UNKNOWN):
+            return 'mdi:eye-off' if self._state == 'on' else 'mdi:eye'
+        if (self._sensor_type == 'streaming_mode' and
+                self._state != STATE_UNKNOWN):
+            return 'mdi:camera' if self._state == 'on' else 'mdi:camera-off'
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the units of measurement."""
+        return SENSOR_TYPES.get(self._sensor_type)[1]
+
+    async def update(self):
+        """Get the latest data and updates the state."""
+        _LOGGER.debug("Pulling data from %s sensor", self._name)
+        await self._camera.update()
+
+        if self._sensor_type == 'last_activity_time':
+            last_activity = await self._camera.last_activity
+            if last_activity is not None:
+                last_activity_time = as_local(last_activity.end_time_utc)
+                self._state = '{0:0>2}:{1:0>2}'.format(
+                    last_activity_time.hour, last_activity_time.minute)
+        else:
+            state = getattr(self._camera, self._sensor_type, STATE_UNKNOWN)
+            if isinstance(state, bool):
+                self._state = 'on' if state is True else 'off'
```

Comment:
Use `STATE_ON` and `STATE_OFF` from const.py. But probably we should instead make a binary sensor platform for that type.

Reply:
The two binary sensors the camera exposes through the sensor platform are:

* streaming mode: determines whether the camera can return video/stills and whether it will record activities (motion, etc)
* privacy mode: determines whether the camera will record activities if they're detected

Neither of these seem to fit neatly into any existing binary sensor class, which is why I opted to make them sensors instead.

Reply:
We can define binary sensors without a specific device class. If a sensor only has a binary state, it belongs with the binary sensor platform.

Reply:
Ah, my bad. As you advised, I'll migrate these sensors to a binary sensor in my next PR.